### PR TITLE
Refactor run.ts with Result pattern and lazy config loading

### PR DESCRIPTION
**Pull request by gbleu**
## Summary

- Extract magic timeout numbers to `TIMING` constants for maintainability
- Add `Result<T>` type with `ok`/`err` helpers for explicit error handling instead of try/catch
- Break down `prepareDrafts` into focused single-responsibility functions
- Implement lazy config loading via Proxy pattern for better testability
- Extract path constants to dedicated `paths.ts` module
- Add `SignatureFormat` type to improve type safety in PDF signing
- Add early config validation at entry point for fail-fast behavior
- Expand test coverage for edge cases and error paths
- Document required environment variables in README

## Test plan

- [ ] Run `bun test` to verify all tests pass
- [ ] Run `./opm` with valid config to verify automation still works
- [ ] Verify config validation fails gracefully with missing required env vars

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/commands/run.ts [hunk 1]
```diff
@@ -246,167 +264,198 @@ async function collectSignedPdfs(
...
-        continue;
-      }
+    const ccField = page.locator('[aria-label="Cc"]').first();
+    await ccField.waitFor({ state: "visible", timeout: TIMING.CC_FIELD });
+    await ccField.click();
+    await page.keyboard.type(ccList);
+    await page.keyboard.press("Tab");
+    await page.waitForTimeout(TIMING.UI_SETTLE);
+
+    return ok(undefined);
```
[the remark this answers is not in the document]
Comment (gbleu) [reply src/commands/run.ts]: Intentional choice - `undefined` is the natural return for void functions in TS. Using `null` would require explicit `null` checks and doesn't add clarity here.

File: src/commands/run.ts [hunk 2]
```diff
@@ -246,167 +264,198 @@ async function collectSignedPdfs(
...
+    .first();
+  await composeBody.waitFor({ state: "visible", timeout: TIMING.ELEMENT_VISIBLE });
 
-    const emailItem = page.locator(`[data-convid="${item.conversationId}"]`);
-    if ((await emailItem.count()) === 0) {
-      console.log(`  -> Email not found in list, skipping`);
-      await takeErrorScreenshot(page, `email-not-found-${idx}`);
-      continue;
-    }
+  return ok(composeBody);
```
[the remark this answers is not in the document]
Comment (gbleu) [reply src/commands/run.ts]: Fixed in 84fd764 - the entire block is now wrapped in try-catch.

File: src/commands/run.ts [hunk 3]
```diff
@@ -24,8 +33,20 @@ export const TIMING = {
...
+  FILE_CHOOSER: 10000,
+  UPLOAD_COMPLETE: 2000,
+  CC_FIELD: 5000,
+  MOVE_MENU: 5000,
 } as const;
 
+export type Result<T> = { ok: true; value: T } | { ok: false; error: string };
+
+const ok = <T>(value: T): Result<T> => ({ ok: true, value });
+const err = <T>(error: string): Result<T> => ({ ok: false, error });
```
[the remark this answers is not in the document]
Comment (gbleu) [reply src/commands/run.ts]: These are intentionally private to this module. If Result becomes shared across the codebase, we'd extract to a dedicated `result.ts` module. YAGNI for now.

File: src/config.ts [hunk 4]
```diff
@@ -1,47 +1,56 @@
...
-export type Config = typeof config;
+  _config = buildConfig(parsed.data);
+  return _config;
+}
+
+export const config: Config = new Proxy({} as Config, {
+  get(_, prop: keyof Config) {
+    const cfg = loadConfig();
+    return cfg[prop];
+  },
```
[the remark this answers is not in the document]
Comment (gbleu) [reply src/config.ts]: No code uses enumeration on `config`. Adding `ownKeys`/`getOwnPropertyDescriptor`/`has` traps adds complexity for a use case that doesn't exist. If we ever need to enumerate config properties, we'll add the traps then.

File: src/commands/run.ts [hunk 5]
```diff
@@ -246,167 +278,242 @@ async function collectSignedPdfs(
...
-        attachBtn.click().then(() =>
+async function attachSignedPdf(
+  page: Page,
+  tmpPath: string,
+): Promise<Result<void>> {
+  try {
+    const attachBtn = page.getByRole("button", { name: "Attach file" });
+    await attachBtn.waitFor({
+      state: "visible",
+      timeout: TIMING.DOWNLOAD_MENU,
```
[the remark this answers is not in the document]
Comment (gbleu) [reply src/commands/run.ts]: Fixed - now uses `TIMING.ELEMENT_VISIBLE` which is semantically correct.

File: src/commands/run.ts [hunk 6]
```diff
@@ -246,167 +278,242 @@ async function collectSignedPdfs(
...
       .first();
-    await composeBody.waitFor({ state: "visible", timeout: 10000 });
+    await composeBody.waitFor({
+      state: "visible",
+      timeout: TIMING.ELEMENT_VISIBLE,
+    });
+    return ok(composeBody);
+  } catch {
+    return err("Compose body not found");
+  }
```
[the remark this answers is not in the document]
Comment (gbleu) [reply src/commands/run.ts]: Fixed - error details now included in Result error message.

File: src/config.ts [hunk 7]
```diff
@@ -1,47 +1,56 @@
...
+});
```
[the remark this answers is not in the document]
Comment (gbleu) [reply src/config.ts]: The Proxy enables testability: tests can mock the entire `config` export before importing modules that use it. Without Proxy, those modules would capture the direct object reference at import time. The early `loadConfig()` call is intentional fail-fast for user experience, separate concern from testability.

File: src/commands/run.ts [hunk 8]
```diff
@@ -246,167 +278,242 @@ async function collectSignedPdfs(
...
+    }
 
-      unlinkSync(tmpPath);
-    } catch (e) {
-      console.error(`  -> Attach failed: ${e}`);
-      await takeErrorScreenshot(page, `attach-fail-${idx}`);
-      try { unlinkSync(tmpPath); } catch {}
-      continue;
+    return ok(undefined);
+  } catch {
```
[the remark this answers is not in the document]
Comment (gbleu) [reply src/commands/run.ts]: Fixed - error details now included.

File: src/commands/run.ts [hunk 9]
```diff
@@ -246,167 +278,242 @@ async function collectSignedPdfs(
...
-    // Wait for draft indicator
-    await page.waitForTimeout(TIMING.CONTENT_LOAD);
+async function moveEmailToInbox(
+  page: Page,
+  conversationId: string,
+): Promise<Result<void>> {
+  const emailInList = page
+    .locator(`[data-convid="${escapeCssValue(conversationId)}"]`)
+    .first();
+  await emailInList.click();
```
[the remark this answers is not in the document]
Comment (gbleu) [reply src/commands/run.ts]: Fixed - entire function now wrapped in try-catch with error details.

File: src/lib/css.ts [hunk 10]
```diff
@@ -0,0 +1,6 @@
+export function escapeCssValue(value: string): string {
+  return value.replace(/[\0-\x1f\x7f"\\'[\]()]/g, (char) => {
```
[the remark this answers is not in the document]
Comment (gbleu) [reply src/lib/css.ts]: The `\XX ` format (hex code + space delimiter) is the correct CSS escape sequence per the CSS spec. The trailing space tells the parser where the hex code ends. This is standard and works correctly in Playwright. Tested with special characters.

File: src/lib/pdf.test.ts [hunk 11]
```diff
@@ -201,28 +225,31 @@ describe("signPdf", () => {
...
-  test("signature file path with .PNG extension is treated as PNG", async () => {
-    const pdfBytes = await Bun.file("test/fixtures/sample.pdf").bytes();
+  test("throws on empty PDF", async () => {
     const sigBytes = await Bun.file("test/fixtures/signature.png").bytes();
-    const position = { x: 100, y: 100, width: 50, height: 20 };
+    const position = { x: 0, y: 0, width: 50, height: 25 };
 
-    // Should work with uppercase extension
-    const result = await signPdf(pdfBytes, sigBytes, "sig.PNG", position);
+    await expect(signPdf(new Uint8Array([]), sigBytes, "png", position)).rejects.toThrow();
```
[the remark this answers is not in the document]
Comment (gbleu) [reply src/lib/pdf.test.ts]: Testing that it throws is sufficient. We don't control pdf-lib's error messages and coupling to them would make tests brittle.

File: src/lib/pdf.test.ts [hunk 12]
```diff
@@ -201,28 +225,31 @@ describe("signPdf", () => {
...
+  });
 
-    expect(result).toBeInstanceOf(Uint8Array);
+  test("throws on invalid PDF", async () => {
+    const invalidPdf = new Uint8Array([1, 2, 3, 4, 5]);
+    const sigBytes = await Bun.file("test/fixtures/signature.png").bytes();
+    const position = { x: 0, y: 0, width: 50, height: 25 };
+
+    await expect(signPdf(invalidPdf, sigBytes, "png", position)).rejects.toThrow();
```
[the remark this answers is not in the document]
Comment (gbleu) [reply src/lib/pdf.test.ts]: Same as above - testing for throw is sufficient without coupling to internal error messages.

File: README.md [hunk 13]
```diff
@@ -41,6 +41,17 @@ To quickly move emails to the folder:
...
+Edit `.env` with your settings. Required variables:
+
+| Variable | Description |
+|----------|-------------|
+| `OPM_MY_EMAIL` | Your email address (to identify your messages) |
+| `OPM_SIGNATURE_X` | Signature X position on page (pixels from left) |
+| `OPM_SIGNATURE_Y` | Signature Y position on page (pixels from bottom) |
+| `OPM_SIGNATURE_WIDTH` | Signature width in pixels |
+| `OPM_SIGNATURE_HEIGHT` | Signature height in pixels |
+| `OPM_REPLY_MESSAGE` | Message text for the reply |
```
[the remark this answers is not in the document]
Comment (gbleu) [reply README.md]: The table lists variables without defaults (will fail validation if missing). Variables with defaults are documented in `.env.example`. This avoids duplication.
